rootbeerlib/src/main/jni/toolChecker.cpp

```diff
@@ -78,6 +80,59 @@ int exists(const char *fname)
 
 
 
+int checkForMagiskUDS()
+{
+    int detect_count = 0;
+    int result = 0;
+
+    // Magisk UDS Detection Method
+    FILE *fh = fopen("/proc/net/unix", "r");
+    if (fh) {
+        for (;;) {
+            char filename[BUFSIZE] = {0};
+            uint32_t a, b, c, d, e, f, g;
+            int count = fscanf(fh, "%x: %u %u %u %u %u %u ",
+                               &a, &b, &c, &d, &e, &f, &g);
+            if (count == 0) {
+                if (!fgets(filename, BUFSIZE, fh)) {
+                    break;
+                }
+                continue;
+            } else if (count == -1) {
+                break;
+            } else if (!fgets(filename, BUFSIZE, fh)) {
+                break;
+            }
+
+            char *ptr = strtok(filename, "@");
+            if(ptr) {
+                if(strstr(ptr, "/")) {
+                    ;
+                } else if(strstr(ptr, " ")) {
+                    ;
+                } else if(strstr(ptr, ".")) {
+                    ;
+                } else {
+                    if (strlen(ptr) >= 32) {
+                        char temp[128] = { 0 };
+                        sprintf(temp, "- [Method] Magisk 루팅이 의심됨!!\n%s\n", ptr);
```

Comment:
Could we translate this to English in keeping with the rest of the library?

Reply:
Korean debugging string was modified in English,
and description of function was added.

Thank you for your confirmation.